starport-core/src/main/scala/com/krux/starport/SummarizeUnmanagedPipelines.scala

```diff
@@ -0,0 +1,35 @@
+package com.krux.starport
+
+import com.krux.starport.util.AwsDataPipeline
+
+object SummarizeUnmanagedPipelines extends StarportActivity {
+
+  def run(): Unit = {
+    val pipelinesInAws = AwsDataPipeline.listPipelineIds() -- inConsoleManagedPipelineIds()
+    val pipelineStatuses = AwsDataPipeline.describePipeline(pipelinesInAws.toSeq: _*)
+
+    logger.info("Unmanaged Pipeline Count By State:")
+    pipelineStatuses.values.groupBy(_.pipelineState).mapValues(_.size).toSeq.sortBy(_._2)(Ordering[Int].reverse)
+      .foreach { case (state, count) => logger.info(s"${state.getOrElse("Unknown")} -> $count") }
+
+    logger.info("Unmanaged Pipeline Count By Date:")
+    pipelineStatuses.values
+      .map { status =>
+        val date = status.creationTime.flatMap(_.split("T").headOption).getOrElse("Unknown")
+        (date, 1)
+      }
+      .groupBy(_._1)
+      .mapValues(_.map(_._2).sum)
```

Comment:
you can do `.map( ... date).groupBy(identity).mapValues(_.size)`

Reply:
Cool. I didn't know of the `identity` function. Thanks!